subsys/bluetooth/host/Kconfig

```diff
@@ -298,6 +298,16 @@ config BT_SMP_SC_ONLY
 	  Security Mode 1 Level 4 stands for authenticated LE Secure Connections
 	  pairing with encryption. Enabling this option disables legacy pairing.
 
+config BT_SMP_UNAUTH_REQUIRE_UNPAIR
```

Comment:
I don't find this name very intuitive. Would you consider `BT_SMP_REJECT_UNAUTH_REPAIR`?

Reply:
I agree with @jhedberg. Either that or something like:

`config BT_SMP_DISABLE_SEC_DOWNGRADE`

Reply:
I guess technically it's not a downgrade, so perhaps:

`config BT_SMP_NO_BOND_OVERWRITE` or similar?

Reply:
> config BT_SMP_DISABLE_SEC_DOWNGRADE

That's slightly different though - this is not a downgrade. It's just that with unauthenticated keys have no way to verifying the identity of the peer, so anyone could "overwrite" the keys we have and start acting like the peer.

Reply:
> config BT_SMP_NO_BOND_OVERWRITE or similar

That's also slightly different 😄  In this case we never had a way to authenticate the peer. With higher security levels you at least have the user interaction that should guarantee that you have physical access to both peers.

Reply:
.. meaning, allowing bond overwrite with higher security levels should not be equally risky

Reply:
I spent some time to come up with the name. I don't really like te term REPAIR, because it seems like something is broken.
`BT_SMP_NO_UNAUTH_OVERWRITE` Since I disallow overwrite when the bond information is unauthenticated. Authenticated pairing can still be overwritten.

Reply:
@joerchan that name works for me